Sources/AsyncAlgorithms/AsyncChannel.swift

```diff
@@ -224,20 +221,19 @@ public final class AsyncChannel<Element: Sendable>: AsyncSequence, Sendable {
           }
         }?.resume()
       }
-      continuation?.resume(with: result)
+      continuation?.resume(returning: element)
     }
   }
   
   /// Send an element to an awaiting iteration. This function will resume when the next call to `next()` is made.
   /// If the channel is already finished then this returns immediately
   public func send(_ element: Element) async {
-      await _send(.success(element))
+    await _send(element)
   }
   
-  /// Send a finish to an awaiting iteration. This function will resume when the next call to `next()` is made.
-  /// If the channel is already finished then this returns immediately
-  public func finish() async {
-      await _send(.success(nil))
+  /// Send a finish to all awaiting iterations.
+  public func finish() {
```

Comment:
making this non async might be a problem from an isolation standpoint. I think that perhaps there is now a scenario where the iteration ends up hanging because a pending nil is awaiting consumption.

Reply:
Could you give an example so I can picture what you have in mind ?

Reply:
so I went over the implementations; this seems like we are in the clear for this.

Reply:
Ok great news. Thanks for the follow up.